### PR TITLE
guest_iommu_test: add a new scenario when ats=off

### PR DESCRIPTION
**Pull request by JinLiul**
Check virtio device function with iommu when disable ats

ID:1542

### CODE REVIEW

**Comment by JinLiul** [comment]
(13/15) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.q35: STARTED
 (13/15) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.q35: PASS (127.03 s)
 (14/15) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.scenario1.q35: STARTED
 (14/15) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.scenario1.q35: PASS (129.91 s)
 (15/15) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.scenario2.q35: STARTED
 (15/15) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.scenario2.q35: PASS (87.22 s)

**Comment by JinLiul** [comment]
Hi @XueqiangWei Could you please help to review. Thanks.

**Comment by XueqiangWei** [comment]
For commit header, it's better to start with the case name, e.g.
guest_iommu_test: add a new scenario when ats=off

We can simply use 'case: Add a new case' format, and in commit body, we can add more introductions on this new case.

File: qemu/tests/cfg/guest_iommu_test.cfg [hunk 1]
```diff
@@ -57,3 +57,6 @@
             smp = 10
         - intremap_off:
             iommu_intremap = off
+        - ats_off:
+            del virtio_dev_ats
+            virtio_dev_ats = off
```
Comment (XueqiangWei) [inline qemu/tests/cfg/guest_iommu_test.cfg:62]: I don't think we need line 61 here, please double check it. Thanks.

**Comment by XueqiangWei** [comment]
By the way, please also add your results with seabios. Thanks.

**Comment by JinLiul** [comment]
test results with seabios:
 (14/16) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.q35: STARTED
 (14/16) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.q35: PASS (92.27 s)
 (15/16) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.scenario1.q35: STARTED
 (15/16) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.scenario1.q35: PASS (92.75 s)
 (16/16) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.scenario2.q35: STARTED
 (16/16) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.scenario2.q35: PASS (72.02 s)

File: qemu/tests/cfg/guest_iommu_test.cfg (hunk 1 above)
Comment (JinLiul) [reply qemu/tests/cfg/guest_iommu_test.cfg:62]: In iommu test loop, virtio_dev_ats = on is a global parameter and other cases don't need this scenario with virtio_dev_ats = off. So I need to del  virtio_dev_ats first to make sure virtio_dev_ats = off works.

**Comment by JinLiul** [comment]
> For commit header, it's better to start with the case name, e.g. guest_iommu_test: add a new scenario when ats=off
> 
> We can simply use 'case: Add a new case' format, and in commit body, we can add more introductions on this new case.

Thanks for your suggestions and I updated the commit info.

File: qemu/tests/cfg/guest_iommu_test.cfg (hunk 1 above)
Comment (XueqiangWei) [reply qemu/tests/cfg/guest_iommu_test.cfg:62]: I double confirmed it, no need line 61 here. 
The results were passed.
 (1/3) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.q35: STARTED
 (1/3) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.q35: PASS (125.72 s)
 (2/3) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.scenario1.q35: STARTED
 (2/3) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.scenario1.q35: PASS (174.77 s)
 (3/3) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.scenario2.q35: STARTED
 (3/3) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.guest_iommu_test.ats_off.scenario2.q35: PASS (156.00 s)

Comment (JinLiul) [reply qemu/tests/cfg/guest_iommu_test.cfg:61]: Tested without line 61 and virtio_dev_ats = off works.Deleted the line 61. Thanks.

**Comment by JinLiul** [comment]
Updated restricted version of qemu-kvm.

**Review by XueqiangWei — APPROVED** [review]

**Comment by XueqiangWei** [comment]
LGTM,
Acked-by: xuwei@redhat.com

**Pull request merged**